pkg/v3/flows/logtrigger.go

```diff
@@ -107,13 +113,18 @@ func newRetryFlow(rs ResultStore, rn ocr2keepersv3.Runner, recoverer Retryer, co
 	return ticker, ticker
 }
 
-type emptyTick struct{}
+type logTick struct {
+	logProvider LogEventProvider
+}
 
-func (et emptyTick) GetUpkeeps(context.Context) ([]ocr2keepers.UpkeepPayload, error) {
-	return nil, nil
+func (et logTick) GetUpkeeps(ctx context.Context) ([]ocr2keepers.UpkeepPayload, error) {
+	if et.logProvider == nil {
+		return nil, nil
+	}
+	return et.logProvider.GetLogs(ctx)
 }
```

Comment:
I like this. It distinguishes the difference between a basic time tick and the log tick source.